### ISSUE TITLE
Replicators

### ISSUE THREAD

**Issue by JacobFischer** [siggame/MegaMinerAI-Game-Ideas#36]
# Story

You are the AI in a Von-Neumann probe, or a self replicating spacecraft. Humanity killed the Earth's ecosystem and it is no longer habitable, so they must move on to other plants. Your home country has sent you out to prepare and investigate potential planets for colonization once the ark ships arrive. But there's one problem, another country's probe arrived in system at around the same time you did. Your home country, which killed the Earth, is stupid and has instructed you to destroy any other probes trying to claim any planets you encounter. Unfortunately they have the same programming and want to destroy you too!

Your goal is to destroy the enemy probe by replicating a functioning combat ship from parts your 3D printers can create. You each only have enough material on board to print 3 ships, so this is the first to win 3 matches (best of 5 rounds).

# Gameplay

![gameplay pic idea](https://i.imgur.com/n3oF46Y.png)

- Simultaneous turn, node based game.
- 5 rounds, first to 3 round victories wins the game
- At the start of each round players start with say 1000 energy
- Both players get **the same** list of potential ShipParts to replicate, each ShipPart costs some energy
- Players build their combat ships by replicating a set of ShipParts and linking them together
- ShipParts either generate power or use power to do things
- Destroy/Incapacitate the enemy player!

# Game Objects

## ShipParts

- These are the "nodes" The have a power draw, which is either negative (draw power) or positive (generate power).
- Power can flow through any non destroyed nodes
- Have a health, when their health reaches 0 they are "destroyed" and power cannot flow through them, and they can't do their action
- Each player starts with 1 "Core" node. It has a ton of health, and if it is destroyed then that ship loses.
- Come with a variety of different types:
    - Weapons: use power, fire missiles, lasers, gauss cannons, etc.
    - Shields: Generate a protective barrier to all nodes connected to them say 3 tiles away. If any of them are hit the shield loses a charge and must recharge that shield, but the hit is canceled.
    - Generator: Generates power to all connected nodes
    - Armor: Nodes that do nothing but have tons of health, thus are hard to kill to stop power from flowing
    - (Option) Engines: let you move 1 unit away or towards the enemy ship. When you are closer your weapons have less time to travel to the enemy, when you are farther they take more time. If we did this weapons would have a linear range.
    - (Optional) Drone Bay: fires a single "drone" unit that flies to the enemy and continually does damage over time. They have to shoot it down or just tank the damage.

 As you can see there are tons of options for different ship parts to strap to the ships.

## Hallway

- A connection (edge) between two Ship Parts (nodes).
- Has a cost following some equation, say expontential
- You want to add edges to many nodes so you can route power around damaged nodes
- Cannot be directly attacked. Think of them more as "glue" between two nodes

## Projectiles

- Weapons that fire things like missiles take some time to travel to their target and hit them, allowing players to try to shoot them down with faster weapons
- projectiles travel at different speeds, e.g. lasers are instant, missiles are slow, gauss shots are medium
- Maybe some can do AoE damage (if they hit)

# Rounds

A round is called when a player's core is destroyed, or if all their weapon parts get destroyed and they have no projectiles in the air (they are incapacitated).

Rounds have 2 stages. The initial stage is the build stage, followed by the combat phase. During combat each turn is simultaneous so you don't know what the enemy does till you both end your turn. Think of each "turn" as 1 sec in real life.

The loser of the round gets to rebuild the next round. **However the winner cannot**, this helps the losing player as they can build to counter the other known ship.

First to win 3 rounds victories wins.

# Pros

- Simultaneous turns are new to use
- Node based, no movement
- Game favors management algorithms rather than pathfinding (our normal roundhouse)
- basically it's a lot of new stuff for us

# Cons

- Balancing ship parts will be difficult
- More abstract than other games
- Node based games might be harder for freshmen without data structures to understand
    - Good candidate for a Spring Semester game
